guardrails/cli/configure.py

```diff
@@ -24,55 +32,69 @@ def save_configuration_file(token: str, no_metrics: bool) -> None:
         rc_file.close()
 
 
+def get_existing_config() -> dict:
+    """Get the configuration from the file if it exists."""
+    home = expanduser("~")
+    guardrails_rc = os.path.join(home, ".guardrailsrc")
+    config = {}
+
+    # If the file exists
+    if os.path.exists(guardrails_rc):
+        with open(guardrails_rc, "r") as rc_file:
+            lines = rc_file.readlines()
+            for line in lines:
+                key, value = line.strip().split("=")
+                config[key] = value
+    return config
+
+
+def _get_default_token() -> str:
+    """Get the default token from the configuration file."""
+    return get_existing_config().get("token", DEFAULT_TOKEN)
+
+
 @guardrails.command()
 def configure(
     token: Optional[str] = typer.Option(
-        help="Your Guardrails Hub auth token.", hide_input=True, default=""
+        default_factory=_get_default_token,
+        help="Your Guardrails Hub auth token.",
+        hide_input=True,
+        prompt="Token (optional)",
     ),
-    no_metrics: Optional[str] = typer.Option(
-        help="Opt out of anonymous metrics collection.", default=False
+    no_metrics: Optional[bool] = typer.Option(
+        DEFAULT_NO_METRICS,
+        "--no-metrics/--metrics",
+        help="Opt out of anonymous metrics collection.",
+        prompt="Disable anonymous metrics reporting?",
+    ),
+    clear_token: Optional[bool] = typer.Option(
+        False,
+        "--clear-token",
+        help="Clear the existing token from the configuration file.",
     ),
 ):
-    """Set the global configuration for the Guardrails CLI and Hub."""
-    try:
-        notice_message = """
-
-    You can find your token at https://hub.guardrailsai.com/tokens
-    """
-        logger.log(level=LEVELS.get("NOTICE"), msg=notice_message)  # type: ignore
-        if not token:
-            token = typer.prompt("Token", hide_input=True)
-        logger.info("Configuring...")
-        save_configuration_file(token, no_metrics)  # type: ignore
+    if clear_token is True:
+        token = DEFAULT_TOKEN
 
+    # Authenticate with the Hub if token is not empty
+    if token != "" and token is not None:
         logger.info("Validating credentials...")
         get_auth()
         success_message = """
+        Login successful.
+
+        Get started by installing our RegexMatch validator:
+        https://hub.guardrailsai.com/validator/guardrails_ai/regex_match
+
+        You can install it by running:
+        guardrails hub install hub://guardrails/regex_match
+
+        Find more validators at https://hub.guardrailsai.com
+        """
+        logger.log(level=LEVELS.get("SUCCESS", 25), msg=success_message)
```

Comment:
This still needs to be wrapped in a try/except

Reply:
done